core/src/main/java/org/elasticsearch/index/query/GeoDistanceQueryParser.java

```diff
@@ -103,7 +104,7 @@ public QueryBuilder fromXContent(QueryParseContext parseContext) throws IOExcept
                     unit = DistanceUnit.fromString(parser.text());
                 } else if ("distance_type".equals(currentFieldName) || "distanceType".equals(currentFieldName)) {
                     geoDistance = GeoDistance.fromString(parser.text());
-                } else if (currentFieldName.endsWith(GeoPointFieldMapper.Names.LAT_SUFFIX)) {
+                } else if (currentFieldName.endsWith(GeoPointFieldMapper.Names.LAT_SUFFIX)) { // TODO IDE complains currentFieldName could be null here - add null check?
```

Comment:
I saw this problem being dealt with in other place by setting currentFieldName to empty String. Worst that can happen then is that it is treated as fieldName in the query, which we should validate later and throw IAE then.